os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -132,11 +160,178 @@ int binary_manager_scan_bootparam(void)
+	bp_info->inuse_idx = latest_idx;
+	bp_info->bp_data = bp_data[latest_idx];
+
+	return BINMGR_OK;
+}
+
+/******************************************************************************
+* Name: binary_manager_update_bpinfo
+*
+* Description:
+*	 This function saves the latest boot parameter information to g_bp_info.
+*
+******************************************************************************/
+int binary_manager_update_bpinfo(void)
+{
+	int ret;
+	binmgr_bpinfo_t bp_info;
+
+	ret = binary_manager_scan_bootparam(&bp_info);
+	if (ret == BINMGR_OK) {	
+		/* Set scanned bootparam data to g_bp_info */
+		g_bp_info.inuse_idx = bp_info.inuse_idx;
+		g_bp_info.bp_data = bp_info.bp_data;
+		bmvdbg("Bootparam[%d] ver: %u, active index: %u, addresses: %x, %x\n", g_bp_info.inuse_idx, g_bp_info.bp_data.version, g_bp_info.bp_data.active_idx, g_bp_info.bp_data.address[0], g_bp_info.bp_data.address[1]);
+	}
+
+	return ret;
+}
+
+/*********************************************************************************
+* Name: binary_manager_update_bootparam
+*
+* Description:
+*	 This function updates new bootparam data in inactive bootparam partition.
+*
+********************************************************************************/
+int binary_manager_write_bootparam(binmgr_bpdata_t *bp_data)
+{
+	int fd;
+	int ret;	
+	int bpdata_size;
+	uint8_t inuse_idx;
+
+	if (!bp_data) {
+		bmdbg("ERROR: Input bp data is NULL\n");
+		return BINMGR_INVALID_PARAM;
+	}
+
+	if (g_bp_info.inuse_idx >= BOOTPARAM_COUNT) {
+		bmdbg("ERROR: Invalid bootparam data, inuse idx %u\n", g_bp_info.inuse_idx);
+		return BINMGR_INVALID_PARAM;
+	}
+
+	fd = binary_manager_open_bootparam();
+	if (fd < 0) {
+		return BINMGR_OPERATION_FAIL;
+	}
+
+	/* Update bootparam data : CRC */
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#if 0 //FIX after bootloader modification
+	bpdata_size = sizeof(binmgr_bpdata_t) - CHECKSUM_SIZE - (((CONFIG_NUM_APPS + 1 - bp_data->app_count) * sizeof(binmgr_userbp_t)) + 1);
+#endif
+	bpdata_size = sizeof(binmgr_bpdata_t) - CHECKSUM_SIZE - (sizeof(bp_data->app_data) + 1);
+#else
+	bpdata_size = sizeof(binmgr_bpdata_t) - CHECKSUM_SIZE;
+#endif
+	bp_data->crc_hash = crc32((uint8_t *)bp_data + CHECKSUM_SIZE, bpdata_size);
+	inuse_idx = g_bp_info.inuse_idx ^ 1;
+
+	ret = lseek(fd, BP_SEEK_OFFSET(inuse_idx), SEEK_SET);
+	if (ret < 0) {
+		bmdbg("ERROR: Seek Failed, errno %d\n", errno);
+		goto errout_with_fd;
+	}
+
+	/* Write bootparam data */
+	ret = write(fd, (FAR uint8_t *)bp_data, sizeof(binmgr_bpdata_t));
+	if (ret != sizeof(binmgr_bpdata_t)) {
+		bmdbg("ERROR: Write Failed, errno %d\n", errno);
+		goto errout_with_fd;
+	}
+	close(fd);
 
 	return BINMGR_OK;
+errout_with_fd:
+	close(fd);
+	return BINMGR_OPERATION_FAIL;
+}
+
+void binary_manager_update_bootparam(int requester_pid, uint8_t type)
+{
+	int ret;
+	int bin_idx;
+	bool need_update;
```

Comment:
Are they needed both `need_update` and `check_update`?
I think we can use one of them.
If needed, please clarify what are the purpose of them.